packages/snap-preact-components/src/components/Molecules/Rating/Rating.tsx

```diff
@@ -6,84 +6,155 @@ import { jsx, css } from '@emotion/react';
-				textAlign: 'center',
-				border: '0px !important',
+			display: 'flex',
+			alignItems: 'center',
+
+			'& .ss__rating__icons': {
+				position: 'relative',
+			},
+
+			'& .ss__rating__stars': {
+				width: '100%',
+				display: 'inline-grid',
+				gridTemplateColumns: '20% 20% 20% 20% 20%',
+
+				'&.ss__rating__stars--full': {
+					position: 'absolute',
+					top: 0,
+					left: 0,
+				},
+
+				'&.ss__rating__stars--empty': {
+					position: 'relative',
+				},
 			},
 
-			'& .fullRatings': {
-				background: `url(${fullRatingSrc || '//4tcdn.blob.core.windows.net/4tjs1/images/allwallstarsfull.png'}) no-repeat`,
-				height: '24px',
-				textAlign: 'left',
-				backgroundRepeat: 'no-repeat',
-				border: '0px !important',
-				float: 'left',
+			'& .ss__rating__stars__star': {
+				overflow: 'hidden',
 			},
 		}),
 };
 
 export const Rating = observer((properties: RatingProps): JSX.Element => {
 	const globalTheme: Theme = useTheme();
 
-	const defaultProps: Partial<RatingProps> = {};
+	const defaultProps: Partial<RatingProps> = {
+		fullIcon: 'star',
+		emptyIcon: 'star-o',
+	};
 
 	const props = mergeProps('rating', globalTheme, defaultProps, properties);
 
-	const { emptyRatingSrc, fullRatingSrc, showEmptyRatings, count, disableStyles, className, style } = props;
+	const { alwaysRender, count, text, disablePartialFill, emptyIcon, fullIcon, disableStyles, className, style, styleScript } = props;
 
-	let rating = props.rating;
+	const subProps: RatingSubProps = {
+		fullIcon: {
+			// default props
+			icon: fullIcon,
+			// global theme
+			...globalTheme?.components?.icon,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+		emptyIcon: {
+			// default props
+			icon: emptyIcon,
+			color: '#ccc',
+			// global theme
+			...globalTheme?.components?.icon,
+			// inherited props
+			...defined({
+				disableStyles,
+			}),
+			// component theme overrides
+			theme: props?.theme,
+		},
+	};
+
+	// normalize values to ensure good rendering
+	let value = props.value;
+	if (isNaN(value)) {
+		value = Number(value) || 0;
+	}
+	if (value < 0) {
+		value = 0;
+	}
+	if (value > 5) {
+		value = 5;
+	}
 
 	const styling: { css?: StylingCSS } = {};
-	if (!disableStyles) {
-		styling.css = [CSS.rating(emptyRatingSrc, fullRatingSrc), style];
+	const stylingProps = { ...props };
+
+	if (styleScript && !disableStyles) {
+		styling.css = [styleScript(stylingProps), style];
+	} else if (!disableStyles) {
+		styling.css = [CSS.rating({ ...props, value }), style];
 	} else if (style) {
 		styling.css = [style];
 	}
 
-	if (isNaN(rating)) {
-		rating = 0;
-	}
-	if (rating > 5) {
-		rating = 5;
-	}
-
-	const star = Math.floor(rating * 20);
+	// with 'disablePartialFill' we are rounding down
+	const numStarsToShow = disablePartialFill ? Math.floor(value) : Math.ceil(value);
```

Comment:
With `disablePartialFill` disabled, `Math.ceil(4.1)` will give you `5`. Do you think using `Math.round` would be better? Or should this be another prop?

Reply:
So the `numStarsToShow` is expected to be 5 when we are doing partial fill. It then will calculate the last star width in the icon map below (using `numStarsToShow` to determine how many _total_ icons to render).

Reply:
If we need to add a new prop to determine the rounding to be used when using `disablePartialFill` (currently we only round down via "floor") we could do that at a future date.